backend/src/test/scala/cromwell/backend/RuntimeAttributeValidationSpec.scala

```diff
@@ -0,0 +1,84 @@
+                               */
+  property("use default and validate it when runtime is not specified") = forAll {
+    (taskName: String, attributeName: String, womValue: WomValue) =>
+      val defaultRuntimeAttributes = Map(attributeName -> womValue)
+
+      val defaultValue = womValue.asWomExpression
+      val validator: Option[WomExpression] => Boolean = {
+        case Some(`defaultValue`) => true
+        case _ => false
+      }
+      BackendWorkflowInitializationActor.validateRuntimeAttributes(taskName, defaultRuntimeAttributes, Map.empty, Map((attributeName,validator)) ).isValid
+  }
+
+  property("return invalid if validator fails the test ") = forAll {
+    (taskName: String, attributeName: String, womValue: WomValue) =>
+      val defaultRuntimeAttributes = Map(attributeName -> womValue)
+
+      BackendWorkflowInitializationActor.validateRuntimeAttributes(taskName, defaultRuntimeAttributes, Map.empty, Map((attributeName,(_: Option[WomExpression]) => false))).isInvalid
+  }
+
+  property("use runtime setting (not default) when both are set") = forAll {
+    (taskName: String, attributeName: String, defaultWomValue: WomValue, runtimeWomExpression: WomExpression) =>
+      val defaultRuntimeAttributes = Map(attributeName -> defaultWomValue)
+      val runtimeAttributes = Map(attributeName -> runtimeWomExpression)
+
+      val validator: Option[WomExpression] => Boolean = {
+        case Some(`runtimeWomExpression`) => true
+        case _ => false
+      }
+      BackendWorkflowInitializationActor.validateRuntimeAttributes(taskName, defaultRuntimeAttributes, runtimeAttributes, Map((attributeName,validator))).isValid
+  }
+
+  property("fail validation if no settings is present but it should be") = forAll {
+    (taskName: String, attributeName: String, defaultWomValue: WomValue, runtimeWomExpression: WomExpression) =>
+
+      val validator: Option[WomExpression] => Boolean = {
+        case None => false
+        case Some(x) => throw new RuntimeException(s"expecting the runtime validator to receive a None but got $x")
+      }
+      BackendWorkflowInitializationActor.validateRuntimeAttributes(taskName, Map.empty, Map.empty, Map((attributeName,validator))).isInvalid
+  }
+
+  property("use the taskName and attribute name in correct places for failures") = forAll {
+    (taskName: String, attributeName: String, defaultWomValue: WomValue, runtimeWomExpression: WomExpression) =>
+
+      val validator: Option[WomExpression] => Boolean = {
+        case None => false
+        case Some(x) => throw new RuntimeException(s"expecting the runtime validator to receive a None but got $x")
+      }
+      BackendWorkflowInitializationActor.validateRuntimeAttributes(taskName, Map.empty, Map.empty, Map((attributeName,validator))).fold(
+        { errors =>
+          val error = errors.toList.head
+          all(
+            "attribute name should be set correctly" |: error.runtimeAttributeName == attributeName,
```

Comment:
my IntelliJ doesn't know what `|:` is and actually neither do I 😢

Reply:
`|:` adds a description to an assertion.